core/src/test/scala/kafka/server/KafkaRequestHandlerTest.scala

```diff
@@ -698,4 +698,81 @@ class KafkaRequestHandlerTest {
     // cleanup
     brokerTopicStats.close()
   }
+
+  @Test
+  def testRequestThreadMetrics(): Unit = {
```

Comment:
How about calling this method something like `testMetricsForMultipleRequestPools`